pennylane/ops/functions/generator.py

```diff
@@ -64,12 +60,17 @@ def _generator_prefactor(gen):
 
     prefactor = 1.0
 
-    if isinstance(gen, Prod):
-        gen = qml.simplify(gen)
+    # simplify
+    gen = qml.simplify(gen) if isinstance(gen, Prod) else gen
 
-    if isinstance(gen, (Hamiltonian, LinearCombination)):
+    if isinstance(gen, LinearCombination):
         gen = qml.dot(gen.coeffs, gen.ops)  # convert to Sum
 
+    # return stuff
+    if isinstance(gen, Prod):
+        coeffs, ops = gen.terms()
+        return ops[0], coeffs[0]
```

Comment:
For some reason this is needed to make the generator for a Controlled op separate the `coeff` and `op` correctly, otherwise it goes from returning `coeff, op` on master to returning `1.0, coeff*op` here. I don't know why.